packages/opentelemetry-core/src/trace/sampler/ProbabilitySampler.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Sampler, SpanContext } from '@opentelemetry/types';
+
+/** Sampler that samples a given fraction of traces. */
+export class ProbabilitySampler implements Sampler {
+  readonly description = 'ProbabilitySampler';
```

Comment:
What's the role of the `description` field? Is that required by the spec?

I asked because it adds a (small) amount of weight to the generated JS, though good bundlers would probably tree shake it out.

Reply:
Yes, this is defined in the specs. This may be displayed on debug pages or in the logs.

Reply:
OK, thanks for explaining - should this one involve the specified probability value?

Reply:
Sure. I would like to do this in separate PR, is it ok? :)

Reply:
Yes, sure!

Reply:
Could this be implemented as the value returned by `sampler.toString()`? Then it would work for `console.log`, JSON, and most serializers out of the box.